Buildings/Fluid/MixingVolumes/MixingVolumeEvaporation.mo

```diff
@@ -0,0 +1,110 @@
+within Buildings.Fluid.MixingVolumes;
+model MixingVolumeEvaporation
+  "Mixing volume for water evaporation at equilibrium"
+  extends Buildings.Fluid.Interfaces.PartialWaterPhaseChange;
+  extends Buildings.Fluid.Interfaces.PartialTwoPortTwoMedium(
+    redeclare final package Medium_a=MediumWat,
+    redeclare final package Medium_b=MediumSte);
+
+  parameter Modelica.SIunits.Volume V "Total volume";
+
+  MediumSte.Temperature T = MediumSte.temperature_phX(
+    p=p, h=bal.hOut, X=MediumSte.X_default)
+    "Temperature of the fluid";
+  Modelica.SIunits.Pressure p = port_a.p
+    "Pressure of the fluid";
+
+  Modelica.Thermal.HeatTransfer.Interfaces.HeatPort_a heatPort(
+    T(start=T_start)) if not steadyDynamics
+  annotation (Placement(transformation(extent={{-10,-90},{10,-110}})));
+  Modelica.Blocks.Interfaces.RealOutput VLiq(unit="m3") "Liquid volume"
+  annotation (Placement(transformation(
+        origin={110,70},
+        extent={{-10,-10},{10,10}},
+        rotation=0), iconTransformation(
+        extent={{-10,-10},{10,10}},
+        rotation=0,
+        origin={110,70})));
+
+protected
+  final parameter Boolean steadyDynamics=
+    if energyDynamics == Modelica.Fluid.Types.Dynamics.SteadyState then true
+      else false
+    "= true, if steady state formulation";
+
+  Buildings.Fluid.Interfaces.ConservationEquationEvaporation bal(
+    redeclare package MediumSte = MediumSte,
+    redeclare package MediumWat = MediumWat,
+    allowFlowReversal=allowFlowReversal,
+    energyDynamics=energyDynamics,
+    massDynamics=massDynamics,
+    p_start=p_start,
+    T_start=T_start,
+    m_flow_nominal=m_flow_nominal,
+    show_T=show_T,
+    V=V)
+    "Energy and mass balance equations with phase change"
+    annotation (Placement(transformation(extent={{20,-10},{40,10}})));
+
+  Buildings.HeatTransfer.Sources.PrescribedTemperature preTem if not steadyDynamics
+    "Port temperature"
+    annotation (Placement(transformation(extent={{-60,-60},{-40,-40}})));
+  Modelica.Blocks.Sources.RealExpression portT(y=T) if  not steadyDynamics
+    "Port temperature"
+    annotation (Placement(transformation(extent={{-90,-60},{-70,-40}})));
```

Comment:
I am a bit confused by these objects. The temperature boundary for that heatport would be the `bal` temperature, which you calculate from the leaving specific enthalpy `hOut` with:
```
  MediumSte.Temperature T = MediumSte.temperature_phX(
    p=p, h=bal.hOut, X=MediumSte.X_default)
    "Temperature of the fluid";
```
The specific enthalpy in turn is calculated from the steam pressure and temperature at saturation with:
```
  state_saturated.p = p;
  state_saturated.T = T;
  T = MediumSte.saturationTemperature(p);
  hSte=MediumSte.specificEnthalpy(state_saturated);
  ...
  hOut = hSte;
```
Isn't it possible to simplify? Could you expose the temperature directly from `ConservationEquationEvaporation` and use that as the prescribed boundary condition? Even better, you could directly specify a heat port in the `ConservationEquationEvaporation` which is directly tied to the volume heat flow and temperature. With that second recommendation, there wouldn't be a need for two separate class since the `MixingVolumeEvaporation` wrapper would be essentially obsolete.